### PR TITLE
Add `Decimal.floor`, `.ceil`, and `.trunc`

### PR DESCRIPTION
**Pull request by GregoryTravis**
### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso [hunk 1]
```diff
@@ -850,6 +850,63 @@ type Decimal
...
+             Decimal.new "12.34" . ceiling
+             # => Decimal.new "13"
+
+       > Example
+         Compute the ceiling of -12.34.
+
+             Decimal.new "-12.34" . ceiling
+             # => Decimal.new "-12"
+    ceil : Integer
+    ceil self -> Decimal = Decimal_Utils.ceil self.big_decimal
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso]: The type annotations are inconsistent.

Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso]:
```suggestion
    ceil self -> Integer = Decimal_Utils.ceil self.big_decimal
```

Comment (GregoryTravis) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso]: Fixed; added test.

File: test/Base_Tests/src/Data/Decimal_Spec.enso [hunk 2]
```diff
@@ -845,6 +845,16 @@ add_specs suite_builder =
...
 
+        group_builder.specify "should return an Integer" <|
+            small = Decimal.new "1.1"
+            big = Decimal.new "1E120" + small
+            small.ceil . should_be_a Integer
+            big.ceil . should_be_a Integer
+            small.floor . should_be_a Integer
+            big.floor . should_be_a Integer
+            small.truncate . should_be_a Integer
+            big.truncate . should_be_a Integer
```
Comment (radeusgd) [inline test/Base_Tests/src/Data/Decimal_Spec.enso]: I think I'd merge this to be part of the tests above, but this is OK too.

**Review by radeusgd — APPROVED** [review]
Looks good

File: test/Base_Tests/src/Data/Decimal_Spec.enso (hunk 2 above)
Comment (GregoryTravis) [reply test/Base_Tests/src/Data/Decimal_Spec.enso]: Merged.

**Review by jdunkerley — APPROVED** [review]

**Pull request merged**